grc/gui/canvas/param.py

```diff
@@ -156,6 +160,21 @@ def format_block_surface_markup(self):
         Returns:
             a pango markup string
         """
+
+        if str(self.value) == str(self._evaluated): # This is not a parameter defined by a variable
+            var_value = Utils.encode(self.pretty_print().replace('\n', ' '))
+        else: # this is defined by a variable, show either the variable name or the value depending on options
+            var_value = None
+
+            if Actions.TOGGLE_SHOW_VARIABLE_PARAMS_NAME.get_active():
+                var_value = "<i>"+Utils.encode( self.truncate(self.value) ) + "</i>"
+
+                if Actions.TOGGLE_SHOW_VARIABLE_PARAMS_VALUE.get_active():
+                    var_value = var_value + "=" + Utils.encode(
+                        self.pretty_print().replace('\n', ' '))
+            else:
+                var_value = Utils.encode( self.pretty_print().replace('\n', ' ') )
+
```

Comment:
The two actions are independently toggled in the GUI, but here one depends on the other. From a users perspective that may be unexpected. 
With the suggestion above (allow only 3 states) the logic here could be flattened to reflect the 3 possible outcomes. It may be useful to sort-out the logic beforehand by defining `show_value`, `show_expression` based on the Actions and the "is literal" check, and then have a flat `if`, `elif` here for output formatting

Reply:
this in excellent idea